app/controllers/reservations_controller.rb

```diff
@@ -138,6 +138,17 @@ def show
     @reservation  = Reservation.find(params[:id])
 
     raise ActiveRecord::RecordNotFound if @reservation != @order_detail.reservation
+
+    respond_to do |format|
+      format.html
+
+      format.ics do
+        calendar = ReservationCalendar.new(@reservation, request.host)
```

Comment:
Yes, I think the `request.host` actually might handle this (my developer one came in as `localhost:3000`.